ironfist/src/cpp/combat/combat.cpp

```diff
@@ -500,11 +500,12 @@ void army::SpecialAttack() {
 			else
 				possibleTarget = target->GetAdjacentCellIndex(target->occupiedHex, i);
 			if (possibleTarget != -1) {
-				if (gpCombatManager->combatGrid[possibleTarget].unitOwner != -1) {
-					army *targ = &gpCombatManager->creatures[gpCombatManager->combatGrid[possibleTarget].unitOwner][gpCombatManager->combatGrid[possibleTarget].stackIdx];
-					if (!gArmyEffected[gpCombatManager->creatures[gpCombatManager->combatGrid[possibleTarget].unitOwner][gpCombatManager->combatGrid[possibleTarget].stackIdx].owningSide][gpCombatManager->creatures[gpCombatManager->combatGrid[possibleTarget].unitOwner][gpCombatManager->combatGrid[possibleTarget].stackIdx].stackIdx]) {
+				hexcell targetHex = gpCombatManager->combatGrid[possibleTarget];
```

Comment:
Uhhh.....do you maybe want to make this a hexcell& or a hexcell* ? Right now you're copying it.